examples/connext_dds/coherent_presentation/c++11/coherent_subscriber.cxx

```diff
@@ -9,76 +9,70 @@
  use the software.
  ******************************************************************************/
 
-#include <cstdlib>
-#include <iostream>
+#include <dds/sub/ddssub.hpp>
+#include <dds/core/ddscore.hpp>
+#include <rti/config/Logger.hpp>  // for logging
 
 #include "coherent.hpp"
-#include <dds/dds.hpp>
-#include <rti/core/ListenerBinder.hpp>
-
-using namespace dds::core;
-using namespace dds::core::policy;
-using namespace dds::core::status;
-using namespace dds::domain;
-using namespace dds::topic;
-using namespace dds::sub;
-using namespace dds::sub::qos;
-using namespace dds::sub::status;
-using namespace rti::core;
-using namespace rti::sub;
+#include "application.hpp"  // for command line parsing and ctrl-c
 
 // Transform a DDS sample into a pair value for a dictionary.
 std::pair<char, int> sample2map(const coherent &data)
 {
     return std::make_pair(data.field(), data.value());
 }
 
-class CoherentListener : public NoOpDataReaderListener<coherent> {
-public:
-    void on_data_available(DataReader<coherent> &reader)
-    {
-        // Take all samples
-        LoanedSamples<coherent> samples = reader.take();
-
-        // Process samples and add to a dictionary
-        std::map<char, int> values;
-        std::transform(
-                rti::sub::valid_samples(samples.begin()),
-                rti::sub::valid_samples(samples.end()),
-                std::inserter(values, values.begin()),
-                sample2map);
-        std::cout << std::endl;
-
-        // Print result
-        std::cout << "Received updates:" << std::endl;
-        for (const auto &value : values) {
-            std::cout << " " << value.first << " = " << value.second << ";";
-        }
-        std::cout << std::endl;
+int process_data(dds::sub::DataReader<coherent> reader)
+{
+    int count = 0;
+    // Take all samples
+    dds::sub::LoanedSamples<coherent> samples = reader.take();
+
+    // Process samples and add to a dictionary
+    std::map<char, int> values;
+    std::transform(
+            rti::sub::valid_samples(samples.begin()),
+            rti::sub::valid_samples(samples.end()),
+            std::inserter(values, values.begin()),
+            sample2map);
```

Comment:
Use a lambda instead of `sample2map`:
```suggestion
            [](const coherent &data) { return std::make_pair(data.field(), data.value()); });
```